### PR TITLE
Support Long type config values

### PR DESCRIPTION
**Pull request by xichen01**
### What changes are proposed in this pull request?

Revert getLong API

### Why are the changes needed?
There are some offline discussions, it is believed that some `propertyKeys` may require a value of type Long, so the revert getLong API is required.

Use case: add a propertykey to limit the capacity of inode, it should be defined as long

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 1]
```diff
@@ -301,6 +301,13 @@ public int getInt(PropertyKey key)
     return (int) get(key);
   }
 
+  @Override
+  public long getLong(PropertyKey key)
+  {
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:306]: This should in the previous line.

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 2]
```diff
@@ -301,6 +301,13 @@ public int getInt(PropertyKey key)
...
+    checkArgument(key.getType() == PropertyKey.PropertyType.INTEGER);
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:307]: It should be `PropertyKey.PropertyType.INTEGER` or `PropertyKey.PropertyType.LONG` ?

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 3]
```diff
@@ -301,6 +301,13 @@ public int getInt(PropertyKey key)
...
+    return (int) get(key);
```
Comment (maobaolong) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:308]: So we should cast it to long?

**Review by maobaolong — CHANGES_REQUESTED** [review]
Left some comment, Thanks for bring the getLong back @xichen01

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 4]
```diff
@@ -2566,7 +2579,7 @@ public String toString() {
           .setScope(Scope.MASTER)
           .build();
   public static final PropertyKey MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES =
-      intBuilder(Name.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES)
+      longBuilder(Name.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES)
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:2582]: I don't think these property keys need to be long. Integer can already go up to 2billion+, which is probably larger than what our system can handle already. If we really need a LONG type property key, let's add that. But these keys should not be of type LONG.

**Review by maobaolong — APPROVED** [review]
It LGTM for this PR, revert the getLong API can make users can use long propertykey. @xichen01 Would you please have a discussion with @rongrong and reach a consensus.

**Comment by xichen01** [comment]
failed UT
```
Error: 3.508 [ERROR] Tests run: 8, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 4.026 s <<< FAILURE! - in alluxio.util.JvmPauseMonitorTest
Error: 3.513 [ERROR] alluxio.util.JvmPauseMonitorTest.testMockedInfoPause  Time elapsed: 1.641 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at alluxio.util.JvmPauseMonitorTest.testMockedInfoPause(JvmPauseMonitorTest.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```

**Comment by xichen01** [comment]
@rongrong PTAL

**Comment by xichen01** [comment]
@rongrong PTAL. Any comments?

**Comment by maobaolong** [comment]
@rongrong I want to add a propertykey to limit the capacity of inode, it should be defined as long.

**Comment by maobaolong** [comment]
@xichen01 Would you like to resolve the ci check issues?

**Review by maobaolong — APPROVED** [review]
It LGTM, thanks for your contribution!

**Review by elega — APPROVED** [review]

**Comment by maobaolong** [comment]
@elega @jiacheliu3 Could you help to merge this PR?

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**